server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -1592,63 +1598,49 @@ public void testSnapshotTrimmedOperations() throws Exception {
     }
 
     /**
-     * this class mimic behaviour of original {@link Translog} and delegated to the original as well
+     * this class mimic behaviour of original {@link Translog}
      */
-    static class TestingTranslog {
+    static class InMemoryTranslog {
         private final List<List<Translog.Operation>> operationsList = new ArrayList<>();
-        private final Translog translog;
 
-        TestingTranslog(Translog translog) {
-            this.translog = translog;
+        InMemoryTranslog() {
             this.operationsList.add(new LinkedList<>());
         }
 
-        void add(Translog.Operation operation) throws IOException {
-            translog.add(operation);
+        void add(Translog.Operation operation) {
             operationsList.get(operationsList.size() - 1).add(operation);
         }
 
-        void rollGeneration() throws IOException {
-            translog.rollGeneration();
+        void rollGeneration() {
             operationsList.add(new LinkedList<>());
         }
 
-        void trimOperations(long belowTerm, long aboveSeqNo) throws IOException {
-            translog.trimOperations(belowTerm, aboveSeqNo);
-            translog.sync();
-
-            for (Iterator<List<Translog.Operation>> opListIt = operationsList.iterator();opListIt.hasNext();) {
-                if (!opListIt.hasNext()) {
-                    // ignore the last one
-                    break;
-                }
-                List<Translog.Operation> operations = opListIt.next();
-                for (Iterator<Translog.Operation> it = operations.iterator(); it.hasNext(); ) {
-                    Translog.Operation op = it.next();
-                    boolean drop = op.primaryTerm() < belowTerm && op.seqNo() > aboveSeqNo;
-                    if (drop) {
-                        it.remove();
+        void trimOperations(long belowTerm, long aboveSeqNo) {
+            operationsList
+                .stream()
+                // handle all expect last one - it has `current` ops
+                .limit(operationsList.size() - 1)
```

Comment:
"Current" ops should never be trimmed due to our assumptions, that said I think it's good not to model that here but rather make sure that the way we generate ops conforms to those assumptions?